docs/proposals/workflow.md

```diff
@@ -0,0 +1,140 @@
+Workflows (aka DAG workflows since jobs are organized in a Direct Acyclic Graph) are ubiquitous in modern [job schedulers](https://en.wikipedia.org/wiki/Job_scheduler), see for example:
+
+* [luigi](https://github.com/spotify/luigi)
+* [ozie](http://oozie.apache.org/)
+* [azkaban](https://azkaban.github.io/)
+
+Most of the [job schedulers](https://en.wikipedia.org/wiki/List_of_job_scheduler_software) offer workflow functionality to some extent.
+
+## Use Cases
+
+* As a user I want to be able to define job chains such that the completion of a given job will trigger other jobs.
+
+## Implementation
+
+The basic idea consists in adding a label selector to the current Job API object. The new selector will determine the parent jobs.  The parent jobs are the jobs current job will depend on. The current job will be scheduled once all the parent jobs will run to completion.
+The strongest point in favor of this approach is the ability to re-use the current `job` implementation. Using a label selector the current `job` will simply become a `graph` of jobs: all the features/controllers that will be built on top of `job` like for example [#11980](https://github.com/kubernetes/kubernetes/issues/11980) and [#17242](https://github.com/kubernetes/kubernetes/issues/17242) will be automatically extended to support workflows. Not sure about https://github.com/kubernetes/kubernetes/issues/16845 and https://github.com/kubernetes/kubernetes/issues/14188.
+A similar approach is implemented in Chronos for [dependent jobs](https://mesos.github.io/chronos/docs/api.html#adding-a-dependent-job).
+
+### API Object
+
+To implement _workflows_ the `Job` API should be modified:
+
+```go
+// JobSpec describes how the job execution will look like.
+
+// These are valid conditions of a job.
+const (
+	// JobComplete means the job has completed its execution.
+	JobComplete JobConditionType = "Complete"
+	JobWaiting  JobConditionType = "Waiting"
+)
+
+type JobSpec struct {
+
+	// Parallelism specifies the maximum desired number of pods the job should
+	// run at any given time. The actual number of pods running in steady state will
+	// be less than this number when ((.spec.completions - .status.successful) < .spec.parallelism),
+	// i.e. when the work left to do is less than max parallelism.
+	Parallelism *int `json:"parallelism,omitempty"`
+
+	// Completions specifies the desired number of successfully finished pods the
+	// job should be run with. Defaults to 1.
+	Completions *int `json:"completions,omitempty"`
+
+	// Selector is a label query over pods that should match the pod count.
+	Selector *PodSelector `json:"selector,omitempty"`
+
+	// Template is the object that describes the pod that will be created when
+	// executing a job.
+	Template api.PodTemplateSpec `json:"template"`
+
+	// Job labels selector to detect parents jobs.
+	ParentSelector map[string]string `json:"parentSelector"`
+}
```

Comment:
I'm thinking you could left out all the contents of JobSpec struct leaving just the newly added field, to make it more readable. Like this:
```
type JobSpec struct {
    ...
    // Job labels selector to detect parents jobs.
    ParentSelector map[string]string `json:"parentSelector"`
}
```